code/game/machinery/igniter.dm

```diff
@@ -33,36 +92,53 @@
 	fire = 100
 	acid = 70
 
+/// turns the igniter on/off
+/obj/machinery/igniter/proc/toggle()
+	on = !( on )
+	if(on)
+		begin_processing()
+	else
+		end_processing()
+	update_appearance()
+
 /obj/machinery/igniter/attack_hand(mob/user, list/modifiers)
 	. = ..()
 	if(.)
 		return
 	add_fingerprint(user)
+	toggle()
 
-	use_power(active_power_usage)
-	on = !( on )
-	update_appearance()
-
-/obj/machinery/igniter/process() //ugh why is this even in process()?
-	if (on && !(machine_stat & NOPOWER) )
-		var/turf/location = loc
-		if (isturf(location))
-			location.hotspot_expose(1000,500,1)
-	return 1
+/// Have to process to ignite any gas that comes in the turf
+/obj/machinery/igniter/process()
+	var/turf/location = loc
+	if(!isturf(location) || !isopenturf(location)) //don't ignite stuff inside walls
+		on = FALSE
+	if(machine_stat & NOPOWER)
+		on = FALSE
+	if(!on)
+		update_appearance()
+		return PROCESS_KILL
 
-/obj/machinery/igniter/Initialize(mapload)
-	. = ..()
-	icon_state = "igniter[on]"
+	location.hotspot_expose(1000, 500, 1)
+	use_power(active_power_usage) //use power to keep the turf hot
 
 /obj/machinery/igniter/update_icon_state()
-	icon_state = "[base_icon_state][(machine_stat & NOPOWER) ? 0 : on]"
+	icon_state = "[base_icon_state][on]"
 	return ..()
 
 /obj/machinery/igniter/connect_to_shuttle(mapload, obj/docking_port/mobile/port, obj/docking_port/stationary/dock)
 	id = "[port.shuttle_id]_[id]"
 
 // Wall mounted remote-control igniter.
 
+/obj/item/wallframe/sparker
+	name = "Sparker WallFrame"
```

Comment:
is it meant to be named that? it's very weird

Reply:
I mean if you have a better name please tell. I'm not very creative

Reply:
Also the guy who named it "Sparker" in the first place should be under fire for this. I am forced to pick up where he left off